From f1df7bb18977ed65b98947efe3b7ea9ac3a2ecd7 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 04 May 2021 14:03:24 +0300
Subject: [PATCH] [SW-41316] preboot: make flexible fw load setting platform specific

Currently CPU_BOOT_DEV_STS0_FW_LD_COM_EN bit is set for all platforms
as part of module_lkd_enable function. This causes issues in greco
LKD load where flexible FW load is not supported. This patch
moves setting of CPU_BOOT_DEV_STS0_FW_LD_COM_EN bit platform specific
and enables for goya and gaudi.

Change-Id: Iaeeb555cd75f965e02cac733a1d1e1fdd45b870b
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 696b9e2..497bc0d 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -418,6 +418,7 @@
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 /* Hard coded each of the gaudi on-chip TS device properties */
diff --git a/goya/common.c b/goya/common.c
index 6a28545..684fef7 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -25,6 +25,7 @@
 #include <preboot/fit.h>
 #include <preboot/thermal.h>
 #include <string.h>
+#include <boot_if.h>
 
 #include "common.h"
 
@@ -222,6 +223,11 @@
 	return mmCPU_TIMESTAMP_CNTCR;
 }
 
+void lkd_dev_specific_setup(void)
+{
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
+}
+
 #define GOYA_ASIC_TS_MAX_COUNT 3
 
 int hl_ts_get_count(void)
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 2da311c..718cb83 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -30,7 +30,6 @@
 
 void module_lkd_enable(void)
 {
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
